src/Common/src/System/Globalization/IdnMapping.Unix.cs

```diff
@@ -11,44 +11,38 @@ private unsafe string GetAsciiCore(char* unicode, int count)
             uint flags = Flags;
             CheckInvalidIdnCharacters(unicode, count, flags, nameof(unicode));
 
-            const int StackAllocThreshold = 512;
-            if (count < StackAllocThreshold)
+            // Each unicode character is represented by 2 ASCII chars
+            // and a "xn--" prefix of length 4
+            int estimatedLength = count * 2 + 4;
```

Comment:
I know I had this in my pseudo-code, but we need to protect against overflow, e.g.
```C#
int estimatedLength = (int)Math.Min(count * 2L + 4, StackallocThreshold);
```